contracts/utils/uint256_checked.cairo

```diff
@@ -0,0 +1,47 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
+from starkware.cairo.common.math import assert_not_zero
+from starkware.cairo.common.uint256 import (
+    Uint256, uint256_add, uint256_sub, uint256_le, uint256_lt
+)
+
+# Adds two integers. 
+# Reverts if the sum overflows.
+func uint256_checked_add{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    } (a: Uint256, b: Uint256) -> (c: Uint256):
```

Comment:
should we also validate `a` and `b` are proper `Uint`s? i.e. use `uint256_check`